src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
```

Comment:
PullProofAI

There is a potential security vulnerability in the `githubAuthCallback` method. When catching errors, the code checks if `error.status` exists and throws an `HttpException` with the error message and status if available. However, this can potentially leak sensitive information about the error to the client. It's safer to provide a generic error message without exposing internal details.

```diff
- throw new HttpException(error.message, error.status);
+ throw new HttpException('An error occurred during authentication.', HttpStatus.BAD_REQUEST);
```



<!-- This is an auto-generated comment by PullProofAI -->